lib/codecs/XsdDateCodec.js

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+export class XsdDateCodec {
+  constructor() {
+  }
+
+  set({value}) {
+    this.value = value;
+  }
+
+  encodeCBOR(gen) {
+    if(typeof this.value !== 'string') {
+      throw new Error('Invalid input, expected value to be of type string');
+    }
+
+    const parsedDate = Date.parse(this.value);
+
+    if(isNaN(parsedDate)) {
+      throw new Error('Invalid input, could not parse value as date');
+    }
+    if(this.value.indexOf('T') > 0) {
+      throw new Error('Invalid input, date includes time component \
+lost of precision expected');
+    }
```

Comment:
```suggestion
    if(this.value.indexOf('T') > 0) {
      throw new Error('Invalid input, date includes time component \
lost of precision expected');
    }
```

We should probably be using a regex here. It's probably fine for the time being.

Reply:
This should get handled at some point during the conversion to cborg. Instead of throwing errors, these codecs should just not do any compression and return the value as-is so it will be stored uncompressed without loss.